.clang-format

```diff
@@ -0,0 +1,84 @@
+# .clang-format
```

Comment:
i thought there was a BasedOnStyle for Linux code, but maybe i'm remembering wrong

i think we largely seem to follow the Linux kernel style, so should we use that to start with ?
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/.clang-format

obviously have to tweak the ForEachMacros, but the rest ?

Reply:
Okay, I will use this as a base.

Reply:
I use the same option for the C language as in https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/.clang-format now.